src/main/kotlin/link/infra/packwiz/installer/ui/gui/GUIHandler.kt

```diff
@@ -166,6 +166,59 @@ class GUIHandler : IUserInterface {
 		return future.get()
 	}
 
+	override fun showUpdateConfirmationDialog(oldVersions: List<Pair<String, String?>>, newVersions: List<Pair<String, String?>>): IUserInterface.UpdateConfirmationResult {
+		assert(newVersions.isNotEmpty())
+		val future = CompletableFuture<IUserInterface.UpdateConfirmationResult>()
+		EventQueue.invokeLater {
+			val oldVersIndex = oldVersions.map { it.first to it.second }.toMap()
+			val newVersIndex = newVersions.map { it.first to it.second }.toMap()
+			val message = StringBuilder()
+			message.append("<html>" +
+					"The modpacks version has been updated.<br>" +
```

Comment:
Maybe state something like "This modpack update uses newer versions of the following:"